### PR TITLE
[0.1] Support Socket Report

### PR DESCRIPTION
**Pull request by SonyPradana**
supported socket serve/report using `reactphp\socket` as server and client.

 why using  library `reactphp\socket`.
- Production ready
- Requires no extensions

### hot to use
1. run serve.
2. setup config to run using socket report (default, no setup require).
3. add/register dump file.
4. run your app normaly and your dump will show in serve console.

### CODE REVIEW

File: src/JsonPrinter.php [hunk 1]
```diff
@@ -122,8 +121,16 @@ protected function send($out)
             return;
         }
 
+        $uri = (string) Config::get('socket.uri', '127.0.0.1:8080');
         $out = $out === false ? '' : $out;
-        (new Style($out))->out(false);
+
+        $connector = new \React\Socket\Connector();
```
Comment (anggerpradana) [inline src/JsonPrinter.php:127]: how about creating a single instance

File: src/Printer.php [hunk 2]
```diff
@@ -40,8 +40,23 @@ protected function send($out)
...
+        $uri        = (string) Config::get('socket.uri', '127.0.0.1:8080');
+        $out        = $out === false ? '' : $out;
+
+        if ($use_socket === false) {
+            echo $out;
+
+            return;
+        }
+
+        $connector = new \React\Socket\Connector();
```
Comment (anggerpradana) [inline src/Printer.php:53]: how about creating a single instance

**Review by anggerpradana — COMMENTED** [review]
make static property to call repeatly (may save memory).

**Review by anggerpradana — APPROVED** [review]

**Comment by SonyPradana** [comment]
ready to merge, but wait for run manual test for socket report using php 8.

**Pull request merged**